pkg/jobs/job_info_storage_test.go

```diff
@@ -358,3 +361,117 @@ func TestAccessorsWithWrongSQLLivenessSession(t *testing.T) {
 		})
 	}))
 }
+
+func TestJobProgressAndStatusAccessors(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+
+	s := serverutils.StartServerOnly(t, base.TestServerArgs{})
+	ctx := context.Background()
+	defer s.Stopper().Stop(ctx)
+
+	idb := s.InternalDB().(isql.DB)
+	r := s.JobRegistry().(*jobs.Registry)
+
+	createJob := func(id jobspb.JobID) *jobs.Job {
+		job, err := r.CreateJobWithTxn(ctx, jobs.Record{Details: jobspb.BackupDetails{}, Progress: jobspb.BackupProgress{}, Username: username.TestUserName()}, id, nil)
+		require.NoError(t, err)
+		return job
+	}
+
+	job1 := createJob(1)
+	job2 := createJob(2)
+
+	before := s.Clock().Now().GoTime()
+
+	t.Run("status", func(t *testing.T) {
+		// Write two kinds of status for j1.
+		require.NoError(t, idb.Txn(ctx, func(ct context.Context, txn isql.Txn) error {
+			return job1.StatusStorage().Write(ctx, txn, "kind1", "one=a")
+		}))
+		require.NoError(t, idb.Txn(ctx, func(ct context.Context, txn isql.Txn) error {
+			return job1.StatusStorage().Write(ctx, txn, "kind2", "two=x")
+		}))
+		// Update one of j1's kinds of status to a new message.
+		require.NoError(t, idb.Txn(ctx, func(ct context.Context, txn isql.Txn) error {
+			return job1.StatusStorage().Write(ctx, txn, "kind1", "one=b")
+		}))
+		// Write a status for j2.
+		require.NoError(t, idb.Txn(ctx, func(ct context.Context, txn isql.Txn) error {
+			return job2.StatusStorage().Write(ctx, txn, "kind1", "one=aa")
+		}))
+
+		after := s.Clock().Now().GoTime()
+
+		var (
+			msg  string
+			kind jobs.JobStatusKind
+			when time.Time
+			err  error
+		)
+
+		require.NoError(t, idb.Txn(ctx, func(ct context.Context, txn isql.Txn) error {
+			msg, kind, when, err = job1.StatusStorage().Latest(ctx, txn, "kind1")
+			return err
+		}))
+		require.Equal(t, jobs.JobStatusKind("kind1"), kind)
+		require.Equal(t, "one=b", msg)
+		require.True(t, before.Before(when) && after.After(when), "%s < %s < %s", before, when, after)
+
+		require.NoError(t, idb.Txn(ctx, func(ct context.Context, txn isql.Txn) error {
+			msg, kind, when, err = job1.StatusStorage().Latest(ctx, txn, "kind2")
+			return err
+		}))
+		require.Equal(t, jobs.JobStatusKind("kind2"), kind)
+		require.Equal(t, "two=x", msg)
+		require.True(t, before.Before(when) && after.After(when), "%s < %s < %s", before, when, after)
+
+		require.NoError(t, idb.Txn(ctx, func(ct context.Context, txn isql.Txn) error {
+			msg, kind, when, err = job2.StatusStorage().Latest(ctx, txn, "kind1")
+			return err
+		}))
+		require.Equal(t, jobs.JobStatusKind("kind1"), kind)
+		require.Equal(t, "one=aa", msg)
+		require.True(t, before.Before(when) && after.After(when), "%s < %s < %s", before, when, after)
+	})
+
+	t.Run("progress", func(t *testing.T) {
+		// Write two fractions updates for j1.
```

Comment:
could you add a case for getting a progress for a job that has yet to write any progress?